specs/merge/beacon-chain.md

```diff
@@ -59,148 +60,138 @@ We define the following Python custom types for type hinting and readability:
 | Name | Value |
 | - | - |
 | `MAX_BYTES_PER_OPAQUE_TRANSACTION` | `uint64(2**20)` (= 1,048,576) |
-| `MAX_EXECUTION_TRANSACTIONS` | `uint64(2**14)` (= 16,384) |
+| `MAX_TRANSACTIONS_PER_PAYLOAD` | `uint64(2**14)` (= 16,384) |
 | `BYTES_PER_LOGS_BLOOM` | `uint64(2**8)` (= 256) |
 
 ## Configuration
 
-Warning: this configuration is not definitive.
+### Merge
 
-### Transition
+*Note*: The configuration value `MERGE_FORK_EPOCH` is not final.
 
 | Name | Value |
 | - | - |
 | `MERGE_FORK_VERSION` | `Version('0x02000000')` |
-| `MERGE_FORK_EPOCH` | `Epoch(18446744073709551615)` **TBD** |
-| `TRANSITION_TOTAL_DIFFICULTY` | **TBD** |
+| `MERGE_FORK_EPOCH` | `Epoch(18446744073709551615)` |
 
 ## Containers
 
 ### Extended containers
 
-*Note*: Extended SSZ containers inherit all fields from the parent in the original
-order and append any additional fields to the end.
-
 #### `BeaconBlockBody`
 
-*Note*: `BeaconBlockBody` fields remain unchanged other than the addition of `execution_payload`.
-
 ```python
 class BeaconBlockBody(phase0.BeaconBlockBody):
+    # Execution
     execution_payload: ExecutionPayload  # [New in Merge]
 ```
 
 #### `BeaconState`
 
-*Note*: `BeaconState` fields remain unchanged other than addition of `latest_execution_payload_header`.
-
 ```python
 class BeaconState(phase0.BeaconState):
-    # Execution-layer
+    # Execution
     latest_execution_payload_header: ExecutionPayloadHeader  # [New in Merge]
 ```
 
 ### New containers
 
 #### `ExecutionPayload`
 
-The execution payload included in a `BeaconBlockBody`.
-
 ```python
 class ExecutionPayload(Container):
-    block_hash: Hash32  # Hash of execution block
+    # Execution block header fields
     parent_hash: Hash32
-    coinbase: Bytes20
+    coinbase: Bytes20  # 'beneficiary' in the yellow paper
     state_root: Bytes32
-    number: uint64
+    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
+    receipt_root: Bytes32  # 'receipts root' in the yellow paper
+    block_number: uint64  # 'number' in the yellow paper
     gas_limit: uint64
     gas_used: uint64
     timestamp: uint64
-    receipt_root: Bytes32
-    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
-    transactions: List[OpaqueTransaction, MAX_EXECUTION_TRANSACTIONS]
+    # Extra payload fields
+    block_hash: Hash32  # Hash of execution block
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_PAYLOAD]
```

Comment:
Why don't we want to follow YP in case of `receipts_root` and `number`?

Reply:
* `receipt_root` vs `receipts_root`: I just reused your term :) Having said that, I believe your choice is the correct one, and is more consistent with the beacon chain spec which does not pluralise (e.g. `deposit_root` vs `deposits_root`).
* `number` vs `block_number`: I just think "number" is pretty vague as anything can be a number. Adding the `block_` prefix clarifies things and is consistent with `block_hash`.

Reply:
I was thinking about removing inconsistency between `receipt_root` on the beacon chain and `receipts_root` on the execution chain which might lead to message decoding bugs. This seems to be the only argument to follow the YP here. Agreed on `block_number`.

Reply:
Curious what @djrtwo thinks here :)

Reply:
My gut is to prioritize naming consistencies like the following:
1. Consistency with beacon chain and execution engine software (to make RPC naming conventions simple/consistent)
2. Consistency of beacon chain naming with itself/other fields
3. Consistency of beacon chain naming with yellow paper

If `receipts_root` is used across all execution engine software today, then I'd slightly prioritize breaking the beacon chain naming convention to be harmonious with the software it is communicating with. If there isn't consistency there, then keep beacon chain naming consistent with itself and make a note about the yellow paper name mismatch

Reply:
I agree with Danny's prioritisation :)

Looks like Geth uses "receipt root" internally (see [here](https://github.com/ethereum/go-ethereum/search?q=receiptroot)) but that an inconsistency was introduced with Rayonism. See [this line](https://github.com/ethereum/go-ethereum/blob/f79cce5de98332e6d54eb298fce85c69d1082ee2/eth/catalyst/api_types.go#L49) which has both `ReceiptRoot` and `json:"receiptsRoot"`.

I believe the way forward here is to go with `receipt_root` which is consistent with Geth and the beacon chain (but inconsistent with the yellow paper).